components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/internal/util/AuthenticatedUserBuilder.java

```diff
@@ -269,16 +271,18 @@ private AbstractUserStoreManager resolveUserStoreManager(AuthenticatedUserData.U
             throws ActionExecutionResponseProcessorException {
 
         AbstractUserStoreManager userStoreManager;
+        String userStoreDomainName;
         try {
             RealmService realmService = AuthenticatorAdapterDataHolder.getInstance().getRealmService();
             int tenantId = IdentityTenantUtil.getTenantId(context.getTenantDomain());
             UserRealm userRealm = (UserRealm) realmService.getTenantUserRealm(tenantId);
             if (userStore != null && StringUtils.isNotBlank(userStore.getName())) {
-                userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager()
-                        .getSecondaryUserStoreManager(userStore.getName());
+                userStoreDomainName = userStore.getName();
             } else {
-                userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager();
+                userStoreDomainName = getDefaultUserStore();
             }
+            userStoreManager = (AbstractUserStoreManager) userRealm.getUserStoreManager()
+                    .getSecondaryUserStoreManager(userStoreDomainName);
```

Comment:
Does this pass even if the user store is primary

Reply:
Yes, tested with local pack with primary